satellitevu/apis/orders.py

```diff
@@ -124,6 +86,23 @@ def submit(self, *, contract_id: Union[UUID, str], item_ids: Union[List[str], st
 
         return self.make_request(method="POST", url=url, json={"item_id": item_ids})
 
+    def _download_request(
+        self,
+        url: str,
+    ):
+        """
+        Request download, handling retries.
+        """
+        while True:
+            response = self.make_request(method="GET", url=url)
+
+            if response.status == 202:
+                sleep(int(response.headers["Retry-After"]))
```

Comment:
would be nice to offer a scaling factor in the download method, i.e. 0.5 means wait only half the Retry-After value. Should default to 1.0